source/generate_source_file.py

```diff
@@ -0,0 +1,209 @@
+# Copyright (C) 2020 and later: Google, Inc.
+
+"""This purpose of this script is to create source .txt files. For statistical
+analysis on distribution of radicals and strokes, see Notebook script
+'Radical-stroke_Index_Analysis.ipynb'.
+Usage:
+    """
+
+import os
+import pandas as pd
+import random
+
+
+def main():
+    # Download Unihan meta data for radical-stroke analysis
+    os.system(' mkdir Unihan')
+    os.system(' curl -O http://unicode.org/Public/UCD/latest/ucd/Unihan.zip')
+    os.system(' apt-get -y install unzip')
+    os.system(' unzip Unihan.zip -d Unihan/')
+    os.system(' rm Unihan.zip')
+
+    # Make sure required package is installed
+    os.system(' python3 -m pip install pandas ')
+
+    # Create intermediate txt file that includes meta data
+    with open('Unihan/Unihan_RadicalStrokeCounts.txt') as f_in:
+        with open('Unihan_RadicalStrokeCounts_kRSKangXi.txt', "w+") as f_out:
```

Comment:
I really think there are no need for you to generate this Unihan_RadicalStrokeCounts_kRSKangXi.txt file. and then read in this file to build CRS_list. Could you just build the CRS_list here directly?
something like
```
    with open('Unihan/Unihan_RadicalStrokeCounts.txt') as f_in:
         for line in f_in:
             if line[0] == '#' or len(line) <= 1:
                 # Skip
                 continue
             if line.split('\t')[1] == 'kRSKangXi':
                 fields = line.split('\t')
                 codepoint, radical, stroke = fields[0], fields[2].split('.')[0], \
                                      fields[2].split('.')[-1][:-1]
                 character = chr(int('0x'+codepoint[2:], 16))
                 radical_number =  int(radical)
                 radical =  number_to_radical[radical_number]
                 radical_stroke_count = number_to_stoke_count[radical_number]
                 remaining_stroke_count =  int(stroke)
                 if ord(character) <= 131072:
                   CRS_list.append(code_point, character, radical_number, radical, \
                                                 radical_stroke_count, remaining_stroke_count)
```